SymlinkUtility.cs

```diff
@@ -7,119 +7,221 @@
-						GUI.Label(r, "<=>", symlinkMarkerStyle);
-				}
-			}
-			catch {}
-		}
-
-		/**
+        private static void OnProjectWindowItemGUI(string guid, Rect r)
+        {
+            try
+            {
+                string path = AssetDatabase.GUIDToAssetPath(guid);
+
+                if (!string.IsNullOrEmpty(path))
+                {
+                    FileAttributes attribs = File.GetAttributes(path);
+
+                    if ((attribs & FOLDER_SYMLINK_ATTRIBS) == FOLDER_SYMLINK_ATTRIBS)
+                        GUI.Label(r, "<=>", symlinkMarkerStyle);
+                }
+            }
+            catch { }
+        }
+
+        /**
 		 *	Add a menu item in the Assets/Create category to add symlinks to directories.
 		 */
-		[MenuItem("Assets/Create/Folder (Symlink)", false, 20)]
-		static void DoTheSymlink()
-		{
-			string sourceFolderPath = EditorUtility.OpenFolderPanel("Select Folder Source", "", "");
-
-			if( sourceFolderPath.Contains(Application.dataPath) )
-			{
-				UnityEngine.Debug.LogWarning("Cannot create a symlink to folder in your project!");
-				return;
-			}
-
-			// Cancelled dialog
-			if( string.IsNullOrEmpty(sourceFolderPath) )
-				return;
-
-			string sourceFolderName = sourceFolderPath.Split(new char[] { '/', '\\' }).LastOrDefault();
-
-			if( string.IsNullOrEmpty(sourceFolderName) )
-			{
-				UnityEngine.Debug.LogWarning("Couldn't deduce the folder name?");
-				return;
-			}
-
-			Object uobject = Selection.activeObject;
-
-			string targetPath = uobject != null ? AssetDatabase.GetAssetPath(uobject) : "";
-
-			if(string.IsNullOrEmpty(targetPath))	
-				targetPath = "Assets";
-
-			FileAttributes attribs = File.GetAttributes(targetPath);
-
-			if( (attribs & FileAttributes.Directory) != FileAttributes.Directory )
-				targetPath = Path.GetDirectoryName(targetPath);
-
-			targetPath = string.Format("{0}/{1}", targetPath, sourceFolderName);
-
-			if( Directory.Exists(targetPath) )	
-			{
-				UnityEngine.Debug.LogWarning(string.Format("A folder already exists at this location, aborting link.\n{0} -> {1}", sourceFolderPath, targetPath));
-				return;
-			}
-
-	#if UNITY_EDITOR_WIN
-			Process cmd = Process.Start("CMD.exe", string.Format("/C mklink /J \"{0}\" \"{1}\"", targetPath, sourceFolderPath));
-			cmd.WaitForExit();
-	#elif UNITY_EDITOR_OSX || UNITY_EDITOR_LINUX
-			// @todo
-	#endif
-			
-			// UnityEngine.Debug.Log(string.Format("Created symlink: {0} <=> {1}", targetPath, sourceFolderPath));
-
-			AssetDatabase.Refresh(ImportAssetOptions.ForceUpdate);
-		}
-	}
-}
+#if UNITY_EDITOR_OSX
+        // Create an absolute symbolic link
+        [MenuItem("Assets/Create/Folder (Symlink)/Absolute", false, 20)]
+        static void SymlinkAbsolute()
+        {
+            Symlink(true);
+        }
+
+        // Create a relative symbolic link
+        [MenuItem("Assets/Create/Folder (Symlink)/Relative", false, 21)]
+        static void SymlinkRelative()
```

Comment:
Would suggest the following just to avoid the extra nesting

```
[MenuItem("Assets/Create/Folder (Absolute Symlink), false, 20)] 
[MenuItem("Assets/Create/Folder (Relative Symlink), false, 21)]         
```

Reply:
Done. I also removed the #if checks since it looks like Windows does use the same syntax for relative paths.